src/protocol/sort/generate_permutation.rs

```diff
@@ -288,7 +288,8 @@ where
         )
         .await?;
 
-        malicious_validator = MaliciousValidator::new(sh_ctx.narrow(&Sort(bit_num)));
+        malicious_validator =
+            MaliciousValidator::new(sh_ctx.narrow(&Sort(bit_num.try_into().unwrap())));
```

Comment:
I still don't understand why we need to cast here.

Reply:
This code will be purged in favour of multi bit so, please ignore it for now